libs/shared/flagd-core/src/lib/parser.ts

```diff
@@ -1,29 +1,40 @@
 import Ajv from 'ajv';
 import { FeatureFlag, Flag } from './feature-flag';
 import flagDefinitionSchema from '../../flagd-schemas/json/flagd-definitions.json';
+import { ParseError } from '@openfeature/core';
 
 const ajv = new Ajv();
 const matcher = ajv.compile(flagDefinitionSchema);
 
 const evaluatorKey = '$evaluators';
 const bracketReplacer = new RegExp('^[^{]*\\{|}[^}]*$', 'g');
 
+const errorMessages = 'invalid flagd flag configuration';
+
 /**
  * Validate and parse flag configurations.
  */
 export function parse(flagCfg: string): Map<string, FeatureFlag> {
-  if (!isValid(JSON.parse(flagCfg))) {
-    throw new Error('invalid flagd flag configurations');
+  try {
+    const transformed = transform(flagCfg);
+    const flags: { flags: { [key: string]: Flag } } = JSON.parse(transformed);
+    const results = matcher(flags);
+    if (!results) {
+      throw new ParseError(errorMessages);
+    }
```

Comment:
I find `matcher` and `results` a bit confusing here. The ajv docs use `validate`( https://ajv.js.org/guide/getting-started.html#basic-data-validation) . Maybe `validator` and `valid` would be clearer?

Reply:
Good catch, fixed.